components/leave_team_modal/leave_team_modal.jsx

```diff
@@ -0,0 +1,121 @@
+import {WebrtcActionTypes} from 'utils/constants.jsx';
+
+class LeaveTeamModal extends React.PureComponent {
+    static propTypes = {
+
+        /**
+         * Current user id.
+         */
+        currentUserId: PropTypes.string.isRequired,
+
+        /**
+         * Current team id.
+         */
+        currentTeamId: PropTypes.string.isRequired,
+
+        /**
+         * hide action
+         */
+
+        onHide: PropTypes.func.isRequired,
+
+        /**
+         * show or hide modal
+         */
+
+        show: PropTypes.bool.isRequired,
+
+        /**
+         * is the user busy in a video call
+         */
+
+        isBusy: PropTypes.bool.isRequired,
+
+        intl: intlShape.isRequired,
+
+        actions: PropTypes.shape({
+
+            /**
+             * An action to remove user from team
+             */
+
+            removeUserFromTeam: PropTypes.func.isRequired,
+
+            /**
+             * An action to toggle the right menu
+             */
+
+            toggleSideBarRightMenu: PropTypes.func.isRequired,
+        }),
+    };
+
+    handleSubmit = (e) => {
+        this.props.onHide();
+
+        if (this.props.isBusy) {
+            WebrtcStore.emitChanged({action: WebrtcActionTypes.IN_PROGRESS});
+            e.preventDefault();
+            return;
+        }
+        this.props.actions.removeUserFromTeam(this.props.currentTeamId, this.props.currentUserId);
+        this.props.actions.toggleSideBarRightMenu();
+    };
+
+    render() {
+        return (
+            <Modal
+                className='modal-confirm'
+                show={this.props.show}
+                onHide={this.props.onHide}
+            >
+                <Modal.Header closeButton={false}>
+                    <Modal.Title>
+                        <FormattedMessage
+                            id='leave_team_modal.title'
+                            defaultMessage='Leave the team?'
+                        />
+                    </Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    <FormattedMessage
+                        id='leave_team_modal.desc'
+                        defaultMessage='You will be removed from all public and private channels.  If the team is private you will not be able to rejoin the team.  Are you sure?'
+                    />
+                </Modal.Body>
+                <Modal.Footer>
+                    <button
+                        type='button'
+                        className='btn btn-default'
+                        onClick={this.props.onHide}
+                    >
+                        <FormattedMessage
+                            id='leave_team_modal.no'
+                            defaultMessage='No'
+                        />
+                    </button>
+                    <button
+                        type='button'
+                        className='btn btn-danger'
+                        onClick={this.handleSubmit.bind(this)}
```

Comment:
No need to bind and should be fine to change into: `onClick={this.handleSubmit}`

Reply:
I had to bind since I am accessing ```this.props``` inside [handleSubmit](https://github.com/mattermost/mattermost-webapp/blob/5de6726bd7c27ea0f24cccf8d22615435aad8432/components/leave_team_modal/leave_team_modal.jsx#L62) . Without binding I am getting props undefined..

Reply:
Kindly let me know if I am missing something